File: js/1967.js
const fs = require('fs');

const inputs = fs.readFileSync('example.txt').toString().trim().split('\n');
const n = +inputs[0];

const adjList = Array(n + 1)
  .fill()
  .map(() => []);
const isVisited = Array(n + 1).fill(false);
let endPoint = 0;
let result = 0;

for (let i = 1; i < n; i++) {
  const [a, b, c] = inputs[i].split(' ').map(Number);

  adjList[a].push([b, c]);
  adjList[b].push([a, c]);
}

isVisited[1] = true;
dfs(1, 0);

init();

isVisited[endPoint] = true;
dfs(endPoint, 0);

console.log(result);

function init() {
  for (let i = 1; i <= n; i++) {
    isVisited[i] = false;
  }
}
function dfs(node, weight) {
  if (result < weight) {
    endPoint = node;
    result = weight;
  }

  for (let i = 0; i < adjList[node].length; i++) {
    const nextNode = adjList[node][i][0];
    const nextWeight = adjList[node][i][1];
    if (!isVisited[nextNode]) {
      isVisited[nextNode] = true;
      dfs(nextNode, weight + nextWeight);
    }
  }
}

File: c++/1967.cpp
#include <iostream>
#include <vector>

using namespace std;

vector<pair<int, int>> node[10001];
bool isVisited[10001];

int n;
int result = 0;
int endPoint = 0;

void init() {
  result = 0;
  for (int i = 1; i <= n; i++) {
    isVisited[i] = false;
  }
}

void dfs(int num, int weight) {
  if (result < weight) {
    result = weight;
    endPoint = num;
  }
  for (int i = 0; i < node[num].size(); i++) {
    int nextNum = node[num][i].first;
    int nextWeight = node[num][i].second;
    if (!isVisited[nextNum]) {
      isVisited[nextNum] = true;
      dfs(nextNum, weight + nextWeight);
    }
  }
}

int main() {
  freopen("example.txt", "r", stdin);
  cin >> n;

  for (int i = 0; i < n - 1; i++) {
    int a, b, c;
    cin >> a >> b >> c;
    node[a].push_back({b, c});
    node[b].push_back({a, c});
  }

  isVisited[1] = true;
  dfs(1, 0);

  init();

  isVisited[endPoint] = true;
  dfs(endPoint, 0);

  cout << result;

  return 0;
}